build-logic/src/main/kotlin/dokkabuild/tasks/GitCheckoutTask.kt

```diff
@@ -121,4 +126,61 @@ abstract class GitCheckoutTask @Inject constructor(
                 .call()
         }
     }
+
+    private fun gitRepoIsUpToDate(): Provider<Boolean> =
+        providers.of(GitRepoIsUpToDate::class) {
+            parameters.repoDir = localRepoDir
```

Comment:
`expectedCommitId` is not configured. Is it expected?

Reply:
No, that's unexpected, thanks!

I wonder why `parameters.expectedCommitId.get()` didn't throw an error?

Reply:
I found the reason: the check always failed.

`RepositoryCache.FileKey.isGitRepository(repoDir)` always returns false, it should be `RepositoryCache.FileKey.isGitRepository(repoDir.resolve(".git")`. 

I fixed it, and added better logging.